sdk/nodejs/dagger/api/api.ts

```diff
@@ -0,0 +1,127 @@
+import { Engine, GraphQLClient, gql } from "../dist/index.js";
+import { ContainerExecArgs } from "./types.js";
+import { queryBuilder, queryFlatten } from "./utils.js"
+
+type QueryTree = {
```

Comment:
If you transform this type into an object, you could include some `utils`  piece of code in it to simplify code and split responsibilities.

For instance, I'm thinking about the conversion to a query string:
```
${item.operation} ${item.args ? `(${args(item)})` : ''} ${q.length - 1 !== index ? '{' : '}'.repeat(q.length - 1)}  
 ```

Reply:
Let's sync and discuss about it